packages/ecmascript/src/Module.ts

```diff
@@ -257,5 +410,104 @@ export interface ModuleElementList {
 		return new ModuleCtor();
 	}
 
+	/**
+	 * Helper function to handle errors when loading/importing an action or module.
+	 * Invokes the error handler set via {@link Module.setModuleErrorHandler} or, if not set,
+	 * via the {@link DefaultModuleErrorHandlers.SYS_ERROR default error handler}.
+	 * @param {string | Error} err - error (message)
+	 * @returns NULL, unless the error handler rethrows the error.
+	 */
+	function onError(error): null {
+		let eh: ModuleErrorHandler | DefaultModuleErrorHandlers = Module[MODULE_ERROR_HANDLER_KEY];
+		if (typeof eh !== "function") {
+			eh = ModuleErrorHandlers[eh] || ModuleErrorHandlers[DefaultModuleErrorHandlers.SYS_ERROR];
+		}
+		eh(error);
+		return null;
+	}
+
+	/**
+	 * Helper function to construct the absolute path of an action or a module
+	 * based on an action name / relative path and (optional) base module name.
+	 * @param {string} path - action name or relative path:
+	 * - may consist of one or more words (comprising of numbers, latin lower/upercase letters, '-' or '_'),
+	 * separated by '.' or '/'
+	 * - may start with './' or any number of '../', but cannot have consecutive separators or end with a separator
+	 * - slashes (/) and backslashes(\) can be used interchangeably
+	 * @param {string} base - base module name:
+	 * - may consist of one or more words (comprising of numbers, latin lower/upercase letters, '-' or '_'),
+	 * separated by '.'
+	 * - may not start or end with '.'
+	 * Note - use of '-' and capital letters is not limited, despite not being recommended.
+	 * @returns {string} path, conforming to the module name convention
+	 * @throws Error if the path is missing/invalid
+	 * @throws Error if a base path is provided but is invalid
+	 * @throws Error if the path is relative and the base path is missing
+	 * @throws Error if the relative path goes too many steps back (via ../) in the base path
+	 */
+
+	function constructAbsolutePath(path: string, base: string): string {
```

Comment:
This needs to be extensively unit tested. What I mean here: ONLY this method needs to be unit tested. With a data provider that will have: path, base, expectedFailure, expected or sth like that